arbutil/finality_data.go

```diff
@@ -0,0 +1,10 @@
+// Copyright 2021-2025, Offchain Labs, Inc.
+// For license information, see https://github.com/nitro/blob/master/LICENSE
+
+package arbutil
+
+type FinalityData struct {
+	FinalizedMsgCount MessageIndex
+	SafeMsgCount      MessageIndex
+	ValidatedMsgCount *MessageIndex
```

Comment:
separate PR: we should also have block-hashes for all of these, to make sure weird race conditions don't make us finalize a reorged-out block

Reply:
This possible inconsistency is more likely to happen with safe blocks than finalized.
It seems that geth doesn't update safe and finalized blocks in case of reorgs BTW.
ConsensusExecutionSyncer will periodically push finality data to execution, so this inconsistency will eventually disappear.

But it makes sense to add those hashes that you mentioned, considering that SetSafe and SetFinalizes will be called holding ExecutionEngine.createBlocksMutex.

I will create a linear task to handle those issues.